### PR TITLE
Update iface_namingmode/test_iface_namingmode.py::TestShowQueue::test_show_queue_counters to work correctly on VOQ chassis

### PR DESCRIPTION
**Pull request by kartik-arista**
Summary:
Fixes 

https://github.com/sonic-net/sonic-mgmt/issues/7875

### Type of change

- Bug fix

### Back port request
- [ x ] 202205

#### What is the motivation for this PR?

Fix above issue.

#### How did you do it?

- Adapted the BUFFER_QUEUE key parsing logic to always use
  the last 2 fields to extract the interface name instead of assuming
  the first 2 fields.

- Ignore non-local system ports from other cards in the validation
  logic.

#### How did you verify/test it?

Verifed that the the test passes on VOQ chassis and another fixed XGS broadcom SKU.

### CODE REVIEW

**Comment by kartik-arista** [comment]
@arlakshm for viz.

File: tests/iface_namingmode/test_iface_namingmode.py [hunk 1]
```diff
@@ -467,21 +467,35 @@ def test_show_queue_counters(self, setup, setup_config_mode, duthosts, enum_rand
...
+        # For the test to be valid, we should have at least one interface selected
+        assert (len(interfaces) > 0)
+
+        intfsChecked = 0
         if mode == 'alias':
             for intf in interfaces:
-                alias = setup['port_name_map'][intf]
+                alias = setup['port_name_map'].get(intf)
+                if not alias:
+                    continue
```
Comment (neethajohn) [inline tests/iface_namingmode/test_iface_namingmode.py:483]: Please make this specific to voq chassis. on non-voq systems, every interface is expected to have alias returned

File: tests/iface_namingmode/test_iface_namingmode.py [hunk 2]
```diff
@@ -467,21 +467,35 @@ def test_show_queue_counters(self, setup, setup_config_mode, duthosts, enum_rand
...
                 assert (re.search(r'{}\s+[U|M]C|ALL\d\s+\S+\s+\S+\s+\S+\s+\S+'
                                   .format(alias), queue_counter) is not None) \
                     and (setup['port_alias_map'][alias] not in queue_counter)
+                intfsChecked += 1
         elif mode == 'default':
             for intf in interfaces:
+                if intf not in setup['port_name_map']:
+                    continue
```
Comment (neethajohn) [inline tests/iface_namingmode/test_iface_namingmode.py:491]: please make this change specific to voq chassis

File: tests/iface_namingmode/test_iface_namingmode.py (hunk 1 above)
Comment (kartik-arista) [reply tests/iface_namingmode/test_iface_namingmode.py:483]: ack. I was not sure of what assumptions to make here. I can add a "voq chassis" check here (and in the other location mentioned below).

File: tests/iface_namingmode/test_iface_namingmode.py (hunk 2 above)
Comment (kartik-arista) [reply tests/iface_namingmode/test_iface_namingmode.py:498]: Fixed in latest.

File: tests/iface_namingmode/test_iface_namingmode.py (hunk 1 above)
Comment (kartik-arista) [reply tests/iface_namingmode/test_iface_namingmode.py]: fixed in latest.

File: tests/iface_namingmode/test_iface_namingmode.py (hunk 2 above)
Comment (neethajohn) [reply tests/iface_namingmode/test_iface_namingmode.py:498]: don't see any change for this code block

**Review by vmittal-msft — APPROVED** [review]

**Comment by kartik-arista** [comment]
@arlakshm @rlhui 

FYI - I re-ran the sonic-mgmt pipeline and it passed. Please advise on how to proceed ahead.

**Comment by rlhui** [comment]
@wangxin , wondering why this is merged even though it's shown "kvmtest-t0" failed?

**Comment by kartik-arista** [comment]
@rlhui  - please read comment above. I re-ran the entire pipeline and it passed.

**Comment by arlakshm** [comment]
Hi @kartik-arista @rlhui, there was one PR checker which was failing. Checked with @wangxin  offline, some of the PR checker pipelines were changed recently, so the older PR checker which failed is still showing as failed. So, it was ok to merge this PR.

**Pull request merged**